### ISSUE TITLE
Bug with Particle Emitter creation in 1.6 branch

### ISSUE THREAD

**Issue by Tryibion** [FlaxEngine/FlaxEngine#1103]
**Issue description:**
When I create a particle emitter asset, it will open the creation window and then after I choose one it looks like it creates the asset, but it does not let me open it or do anything with it.

![image](https://github.com/FlaxEngine/FlaxEngine/assets/71274967/d0a43c16-2084-46cd-937c-5f90c1a81d79)

**Steps to reproduce:**
Create a particle emitter asset.

**Flax version:**
1.6.6340

**Comment by mafiesto4** [comment]
I was unable to reproduce it. do you have a log with some errors during that session?

**Comment by Tryibion** [comment]
`[ 00:00:16.898 ]: [Info] Cloning asset 'Z:\Flax/Source/FlaxEngine/Content/Editor/Particles/Constant Burst.flax' to 'C:\Users/mrcha/Documents/Flax Projects/Testing16/Content/Particles/Particle Emitter 0.flax'(7ff6e0ee4c6d35997b53b8806575459b).
[ 00:00:16.902 ]: [Info] Changing asset 'bf3bb69847fb03b794370f91693483f2' id to '7ff6e0ee4c6d35997b53b8806575459b' (storage: 'C:\Users/mrcha/AppData/Local/Temp/c561f7cd-44e6-0da4-ca76-839181c3413e/ec27c05a-4d51-72f0-30d3-6a8e29514e72')
[ 00:00:16.902 ]: [Info] Creating package at 'C:\Users/mrcha/AppData/Local/Temp/c561f7cd-44e6-0da4-ca76-839181c3413e/ec27c05a-4d51-72f0-30d3-6a8e29514e72'. Silent Mode: false
[ 00:00:16.902 ]: [Info] File 'C:\Users/mrcha/AppData/Local/Temp/c561f7cd-44e6-0da4-ca76-839181c3413e/ec27c05a-4d51-72f0-30d3-6a8e29514e72' is in use. Trying to release handle to it.
[ 00:00:16.905 ]: [Warning] Invalid asset magic code in Asset 'C:\Users/mrcha/Documents/Flax Projects/Testing16/Content/Particles/Particle Emitter 0.flax'
[ 00:00:16.905 ]: [Error] Failed to load C:\Users/mrcha/Documents/Flax Projects/Testing16/Content/Particles/Particle Emitter 0.flax.
[ 00:00:16.906 ]: [Warning] Cannot copy file to destination.
[ 00:00:16.907 ]: [Warning] Failed to import Settings to C:\Users/mrcha/Documents/Flax Projects/Testing16/Content/Particles/Particle Emitter 0.flax
`

**Comment by Tryibion** [comment]
And then when I try to delete it:

```
[ 00:01:10.061 ]: [Warning] Exception has been thrown during Window.OnKeyDown. The process cannot access the file 'C:\Users\mrcha\Documents\Flax Projects\Testing16\Content\Particles\Particle Emitter 0.flax' because it is being used by another process.
Stack strace:
   at System.IO.FileSystem.DeleteFile(String fullPath)
   at FlaxEditor.Modules.ContentDatabaseModule.Delete(ContentItem item) in Z:\Flax\Source\FlaxEngine\Source\Editor\Modules\ContentDatabaseModule.cs:line 697
   at FlaxEditor.Windows.ContentWindow.Delete(List`1 items) in Z:\Flax\Source\FlaxEngine\Source\Editor\Windows\ContentWindow.cs:line 524
   at FlaxEditor.Content.GUI.ContentView.<.ctor>b__47_1() in Z:\Flax\Source\FlaxEngine\Source\Editor\Content\GUI\ContentView.cs:line 190
   at FlaxEditor.Options.InputActionsContainer.Process(Editor editor, Control control, KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Editor\Options\InputBinding.cs:line 477
   at FlaxEditor.Content.GUI.ContentView.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Editor\Content\GUI\ContentView.cs:line 632
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEditor.GUI.Docking.DockWindow.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Editor\GUI\Docking\DockWindow.cs:line 476
   at FlaxEditor.Windows.EditorWindow.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Editor\Windows\EditorWindow.cs:line 195
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.GUI.ContainerControl.OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\UI\GUI\ContainerControl.cs:line 1162
   at FlaxEngine.Window.Internal_OnKeyDown(KeyboardKeys key) in Z:\Flax\Source\FlaxEngine\Source\Engine\Platform\Window.cs:line 212
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerNoRet1`2.MarshalAndInvoke(Object delegateContext, ManagedHandle instancePtr, IntPtr paramPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Invoker.cs:line 256
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(ManagedHandle instanceHandle, ManagedHandle methodHandle, IntPtr paramPtr, IntPtr exceptionPtr) in Z:\Flax\Source\FlaxEngine\Source\Engine\Engine\NativeInterop.Unmanaged.cs:line 675
[ 00:01:10.061 ]: [Error] Exception has been thrown during Window.OnKeyDown.
```

**Comment by TheJustLink** [comment]
I confirm, the same thing.

**Comment by mafiesto4** [comment]
Maybe you have an invalid Git LFS file of the particle emitter template?  Because it fails on file data reading. Can you open particle emitters from `Content/Editor/Particles` folder inside Flax directory?

![image](https://github.com/FlaxEngine/FlaxEngine/assets/3464853/4220a970-2c4e-4d40-8217-de915ebfa1de)

**Comment by mafiesto4** [comment]
Bug with removing invalid asset was fixed here: a3ab7cd14e66456f006cc71f07ebaa2309ea2958

**Comment by Tryibion** [comment]
> Maybe you have an invalid Git LFS file of the particle emitter template?  Because it fails on file data reading. Can you open particle emitters from `Content/Editor/Particles` folder inside Flax directory?
> 
> ![image](https://github.com/FlaxEngine/FlaxEngine/assets/3464853/4220a970-2c4e-4d40-8217-de915ebfa1de)
> 

Yes I can open the editor templates just fine. I should also note that it does let me create a new empty particle emitter, but it won't let me create one from the templates.

**Comment by schmidt-florian** [comment]
I have the same Issue. Note: Sparks do not work most time. Others work most time. It is something with Sparks I think.

Here is the Log file... I tried it like 25 times (?) before it suddenly worked.
[log.txt](https://github.com/FlaxEngine/FlaxEngine/files/13205946/log.txt)

You can watch this Video, in the last seconds it suddenly worked:


https://github.com/FlaxEngine/FlaxEngine/assets/14238126/3cdafc91-a074-497a-aaee-1b6bc02fa677

**Comment by mafiesto4** [comment]
Fixed in #2418
